### PR TITLE
chore: Removes MongoDB 5 test

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Removes MongoDB 5 test as it's EOL.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 1]
```diff
@@ -287,37 +287,6 @@ func TestAccClusterAdvancedCluster_advancedConfig(t *testing.T) {
...
-		}
-	)
-
-	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:                 acc.PreCheckBasicSleep(t, nil, projectID, clusterName),
-		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
-		CheckDestroy:             acc.CheckDestroyCluster,
-		Steps: []resource.TestStep{
-			{
-				Config: configAdvanced(projectID, clusterName, processArgs, nil, conversion.StringPtr("5")),
```
Comment (EspenAlbert) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go:314]: looks like the final argument is only used in this test (other tests it is nil)
Can we remove it from the configAdvanced func?

**Review by EspenAlbert — APPROVED** [review]
LGTM. Quick comment

File: internal/service/advancedcluster/resource_advanced_cluster_test.go (hunk 1 above)
Comment (lantoli) [reply internal/service/advancedcluster/resource_advanced_cluster_test.go:314]: thx, removed in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2768/commits/a97479cea2ff0f1d6dff023c14fe6a6608ffb332

File: internal/service/advancedcluster/resource_advanced_cluster_test.go [hunk 2]
```diff
@@ -276,48 +276,17 @@ func TestAccClusterAdvancedCluster_advancedConfig(t *testing.T) {
...
 			{
-				Config: configAdvanced(projectID, clusterNameUpdated, processArgsUpdated, conversion.IntPtr(100), nil),
+				Config: configAdvanced(projectID, clusterNameUpdated, processArgsUpdated, conversion.IntPtr(100)),
 				Check:  checkAdvanced(clusterNameUpdated, "TLS1_2", "100"),
 			},
 		},
 	})
 }
 
-func TestAccClusterAdvancedCluster_advancedConfig_MongoDBVersion5(t *testing.T) {
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/resource_advanced_cluster_test.go:290]: q: I see we initially added this test to ensure new attributes not compatible with older MongoDB versions would not cause errors (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2592). Should we rename this test to OldMongoDBVersion and transition over to version 6?

Comment (lantoli) [reply internal/service/advancedcluster/resource_advanced_cluster_test.go:290]: in that link it looks like it's specific to 5.  "set this attribute if MongoDB version is less than 6.0."

Comment (lantoli) [reply internal/service/advancedcluster/resource_advanced_cluster_test.go:290]: I think we can go ahead and delete this test, if compatible checks are not needed for 6

**Pull request merged**